lib/element.ts

```diff
@@ -230,7 +230,7 @@ export class ElementArrayFinder extends WebdriverWebElement {
    *     of element that satisfy the filter function.
    */
   filter(
-      filterFn: (element: ElementFinder, index?: number) => boolean |
+      filterFn: (element: ElementFinder, index: number) => boolean |
```

Comment:
Why does the `index` parameter should be required?

Reply:
Removing the question mark does not mean that you are forced to accept `index` as a parameter in your callback. This is the whole point of https://github.com/angular/protractor/issues/4634.

Reply:
Gotcha! Thanks a lot!